### PR TITLE
Optimised Cypher queries in Neo4j example

### PR DESCRIPTION
**Pull request by adam-cowley**
Thank you for including a Neo4j example!

This PR rewrites the Cypher queries using a [pattern comprehension](https://neo4j.com/docs/cypher-manual/current/syntax/lists/#cypher-pattern-comprehension) which will make them more performant and removes the need for the `collect(DISTINCT ...)` step.

### CODE REVIEW

File: examples/with-neo4j/pages/movie/[title].js [hunk 1]
```diff
@@ -33,13 +33,23 @@ export default function Movie() {
...
           </div>
           <div className="actors">
             <h2>Actors</h2>
             {data.movie.actors.map((actor) => (
-              <div key={actor}>{actor}</div>
+              <div key={actor}>
+                <Link
+                  key={actor}
+                  href="/actor/[name]"
+                  as={{
```
Comment (leerob) [inline examples/with-neo4j/pages/movie/[title].js:46]: You don't need to use `as` anymore - can just use this for the href! 😄

**Review by leerob — COMMENTED** [review]
LGTM after that!

File: examples/with-neo4j/pages/movie/[title].js (hunk 1 above)
Comment (adam-cowley) [reply examples/with-neo4j/pages/movie/[title].js]: OK great, consider it done!

**Comment by adam-cowley** [comment]
I realised a mistake with how the numbers were handled in the front end.  There is a gotcha when it comes to [JavaScript integers and Neo4j (Java) integers](https://github.com/neo4j/neo4j-javascript-driver#numbers-and-the-integer-type).  

Using `disableLosslessIntegers` with instantiating the driver solves the problem in this use case.  In 90% of cases, this workaround will be fine, if the integer happens to be a BigInteger in this case then the number is converted to a string.  Either way it'll be handled the same way by the UI.

**Review by leerob — DISMISSED** [review]

**Review by ijjk — APPROVED** [review]
Thanks for the PR!

**Pull request merged**